### DISCUSSION TITLE
Change DPI but keep image size

### DISCUSSION THREAD

**Discussion by EamonnMR** [ImageMagick/ImageMagick#3124] [Help] [answered]
I've got a bunch of images I'm converting into a PDF like so:
`convert *.jpg "${PWD##*/}.pdf"`
When I resize by 25% the file size goes way down (from ~800mb to 34mb), which is great!
`convert *.jpg -resize 25% "quarter_${PWD##*/}.pdf"`
I suspect that this is because it's re-encoding the images, but it's also a much smaller image. Only problem is it also changes the size of the image! Is there a way to do the same operation as `resize 25%` without resizing, just changing the image density? I would expect something like `-density 25%` to do it, but it does not. Is there such a command, or do I need to figure out the input DPI and specify an exact output DPI? (and if that's the only way to do it, how's it done?)

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform?

If you use IM v7, with "magick", you can change the density with a %[fx:...] expression, like this:
```
magick toes.png -density %[fx:resolution.x*5] x.png
```
This multiplies the horizontal density by 5.

**Reply by EamonnMR** [reply]
I'm on Ubuntu. I built the latest ImageMagick from github and tried this:
```magick *.jpg -density %[fx:resolution.x*5] "${PWD##*/}.pdf"```
but it crashes my entire machine after the 40th image (according to `-verbose`)

**Comment by snibgo** [comment]
"-density" changes a field in metadata. I can't see how it could cause a crash. If you remove that, does it still crash? What happens when you process just a single input jpg, or a subset, instead of all of them?

**Reply by EamonnMR** [reply]
I tried a single image and it didn't crash. I assume it's exhausting some resource. My goal is to lower the quality of the image while keeping the size the same. But you can combine resize with density change to do it! I'll update this to indicate if it worked or crashed, but here's the updated command:
`magick *.jpg -resize 20% -density %[fx:resolution.x/5] "density-resize-all${PWD##*/}.pdf"`

Edit: As discussed elsewhere, this still crashed. See the bash script for what ultimately worked.

**Comment by fmw42** [comment]
You are loading all your images into RAM and then processing them all at the same time. That takes a lot of resources. You should write a script loop and process them one at a time. Or you can use Magick mogrify to process all images in a folder automatically one image at a time. See https://legacy.imagemagick.org/Usage/basics/#mogrify

**Reply by EamonnMR** [reply]
Here's a bash script that does the thing:
```
#!/bin/bash
set -eo pipefail

for file in *.jpg; do
  magick $file -resize 20% -density %[fx:resolution.x/5] $file.tmp;
done

magick *.tmp "${PWD##*/}.pdf";

rm *.tmp;
```

**Comment by snibgo** [comment]
@EamonnMR: In your command, "-density" has no effect on quality or size (in bytes). You could use "-quality X" where X is 20 or 40 or whatever. This will reduce visual quality and size in bytes, without changing size in pixels.
